Framework/Core/src/DataDescriptorQueryBuilder.cxx

```diff
@@ -236,29 +235,69 @@ std::vector<InputSpec> DataDescriptorQueryBuilder::parse(char const* config)
 
 DataDescriptorQuery DataDescriptorQueryBuilder::buildFromKeepConfig(std::string const& config)
 {
-  static const std::regex specTokenRE(R"re((\w{1,4})/(\w{1,16})/(\d*))re");
-  static const std::regex delimiter(",");
+  static const std::regex delim(",");
 
+  std::sregex_token_iterator end;
   std::sregex_token_iterator iter(config.begin(),
                                   config.end(),
-                                  delimiter,
+                                  delim,
                                   -1);
-  std::sregex_token_iterator end;
 
-  std::unique_ptr<DataDescriptorMatcher> result;
+  std::unique_ptr<DataDescriptorMatcher> next, result;
 
   for (; iter != end; ++iter) {
     std::smatch m;
     auto s = iter->str();
-    std::regex_match(s, m, specTokenRE);
-    std::unique_ptr<DataDescriptorMatcher> next;
-    auto newNode = std::make_unique<DataDescriptorMatcher>(
-      DataDescriptorMatcher::Op::And,
-      OriginValueMatcher{m[1]},
-      std::make_unique<DataDescriptorMatcher>(
-        DataDescriptorMatcher::Op::And,
-        DescriptionValueMatcher{m[2]},
-        SubSpecificationTypeValueMatcher{m[3]}));
+    auto newNode = buildNode(s);
+
+    if (result.get() == nullptr) {
+      result = std::move(newNode);
+    } else {
+      next = std::move(std::make_unique<DataDescriptorMatcher>(DataDescriptorMatcher::Op::Or,
+                                                               std::move(result),
+                                                               std::move(newNode)));
+      result = std::move(next);
+    }
+  }
+
+  return std::move(DataDescriptorQuery{{}, std::move(result)});
+}
+
+DataDescriptorQuery DataDescriptorQueryBuilder::buildFromExtendedKeepConfig(std::string const& config)
+{
+  // extended config string consists of ','-separated items
+  // each item consists of 3 parts which are separated by a ':'
+  // "origin/description/subSpec:filename:col1/col2/col3, ... "
+  // here only the first part of each item is extracted, namely origin/description/subSpec
```

Comment:
This should be documented in the Framework/Core/ANALYSIS.md file

Reply:
is done